src/ice/mod.rs

```diff
@@ -854,6 +854,59 @@ mod test {
         }));
     }
 
+    #[test]
+    fn changed_timing_config_takes_effect_immediately() {
+        let _guard = tracing_subscriber::fmt()
+            .with_env_filter("trace")
+            .with_test_writer()
+            .set_default();
+
+        const IDLE_RTO: Duration = Duration::from_secs(60);
+        const NORMAL_RTO: Duration = Duration::from_secs(3);
+
+        let mut a1 = TestAgent::new(info_span!("L"));
+        let mut a2 = TestAgent::new(info_span!("R"));
+
+        let c1 = host("1.1.1.1:1000", "udp");
+        a1.add_local_candidate(c1.clone());
+        a2.add_remote_candidate(c1);
+
+        let c2 = host("2.2.2.2:1000", "udp");
+        a2.add_local_candidate(c2.clone());
+        a1.add_remote_candidate(c2);
+
+        a1.set_controlling(true);
+        a2.set_controlling(false);
+
+        // loop until we're connected.
+        loop {
+            if a1.state().is_connected() && a2.state().is_connected() {
+                break;
+            }
+            progress(&mut a1, &mut a2);
+        }
+
+        // Move to "idle" mode
+        a1.set_initial_stun_rto(IDLE_RTO);
+        a1.set_max_stun_rto(IDLE_RTO);
+        a2.set_initial_stun_rto(IDLE_RTO);
+        a2.set_max_stun_rto(IDLE_RTO);
+
+        // Spin for a bit
+        for _ in 0..10 {
+            progress(&mut a1, &mut a2);
+        }
+
+        let now = a1.poll_timeout().unwrap() - IDLE_RTO;
+
+        a1.set_initial_stun_rto(NORMAL_RTO);
+        a1.set_max_stun_rto(NORMAL_RTO);
+
+        let timeout_after = a1.poll_timeout().unwrap();
+
+        assert_eq!(timeout_after, now + NORMAL_RTO);
```

Comment:
I wasn't sure what the best assertion here is. Calculating `now` based on the timeout and IDLE_RTO is "insider" knowledge about the ICE agent.

An alternative to this would be to calculate the difference between the two `poll_timeout` calls. That has to be the difference between the two RTOs.

Reply:
I suppose it works until it doesn't. Make a code comment about it being dodgy so when it breaks we don't have to scratch our heads.